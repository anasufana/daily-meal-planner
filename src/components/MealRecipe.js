import React from 'react';
import '../css/components/MealRecipe.css';
import RecipeStep from './RecipeStep';

const MealRecipe = props => (
  <div className="recipe-page-container">
    <h1 className="recipe-title-page">{ props.details.recipeTitle }</h1>
    <div className="section-ingredients">
      <div className="recipe-img-box">
        <img className="image-recipe-page" src={`https://spoonacular.com/recipeImages/${props.details.recipeImage}`} alt="test-img" />
        <div className="clock-div-container">
          <div className="clock-div-button">
            <img className="img-clock" src="/img/clock.svg" alt="Time" title="Time" height="20" width="20" />
            <h3 className="recipe-duration-page">{ props.details.readyInMinutes }'</h3>
          </div>
        </div>
      </div>
      <div className="ingredients-box">
        <h2 className="ingredients-title">Ingredients:</h2>
        <ul className="ingredients-list">
          {
             props.details.ingredients[0].map((ingredient) => {
               return (<li className="ingredients-list-item">{ ingredient }</li>)
           })
        }
        </ul>
      </div>
    </div>
    <div>
      {
      props.details.recipes[0].map((recipe, i) => {
        return (
          <div>
            <h1>{recipe.name ? recipe.name : 'Recipe Steps'}</h1>
            {
              recipe.steps.map((step, j) => {
                    return (<RecipeStep key={`${i}${j}`} details={step} />)
              })
            }
          </div>
        )
        })
      }
    </div>
  </div>
);

export default MealRecipe;
