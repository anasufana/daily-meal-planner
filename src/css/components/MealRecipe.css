.recipe-page-container {
  width: 55%;
  margin: 0 auto;
  text-align: left;
  background-color: white;
  padding: 1em;
  border-radius: 5px; }

.recipe-title-page {
  padding: 1em; }

.image-recipe-page {
  width: 100%;
  height: 30vw;
  object-fit: cover;
  box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.1); }

.section-ingredients {
  width: 100%;
  display: grid;
  grid-template-columns: 2fr 1fr;
  grid-column-gap: 30px;
  background-color: beige;
  padding: 2em 2em;
  border-radius: 5px; }

.ingredients-title,
.step-title {
  padding-left: 30px;
  padding-bottom: 20px; }

.step-title {
  padding-top: 30px; }

.ingredients-box {
  border-left: 1px solid rgba(0, 0, 0, 0.2); }

.ingredients-list {
  padding-left: 40px; }

.ingredients-list-item {
  padding-bottom: 3px;
  font-size: 1.2rem; }

.recipe-img-box {
  position: relative; }

.clock-div-container {
  position: absolute;
  top: 20px;
  left: 20px; }

.clock-div-button {
  position: relative; }

.img-clock {
  left: 20px;
  top: 12px;
  position: absolute; }

.recipe-duration-page {
  padding: 10px 30px 10px 50px;
  background-color: rgba(132, 221, 113, 0.8);
  border-radius: 5px;
  border: 2px solid #000; }

.step-description {
  margin-left: 30px;
  line-height: 28px;
  width: 85%;
  font-size: 1.2em;
  padding-bottom: 20px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.3); }

.h1-recipe-steps {
  margin-top: 20px;
  font-size: 1.5em;
  background: rgba(132, 221, 113, 0.8);
  display: inline-block;
  margin-left: 30px;
  padding: 10px;
  border-radius: 5px; }

@media only screen and (max-width: 1440px) and (min-width: 750px) {
  .recipe-page-container {
    width: 70%; }
  .ingredients-title {
    padding-top: 20px;
    padding-left: 20px; }
  .image-recipe-page {
    width: 100%;
    height: 40vw;
    -o-object-fit: cover;
    object-fit: cover; } }

@media screen and (max-width: 750px) {
  .recipe-page-container {
    width: 90%; }
  .section-ingredients {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr;
    background-color: beige;
    padding: 2em 2em; }
  .ingredients-title {
    padding-top: 20px;
    padding-left: 20px; }
  .ingredients-box {
    border-left: none; }
  .image-recipe-page {
    width: 100%;
    height: 70vw;
    -o-object-fit: cover;
    object-fit: cover; } }
