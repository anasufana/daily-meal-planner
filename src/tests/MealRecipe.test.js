/* global it, expect, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import MealRecipe from '../components/MealRecipe';

const details = {
  details: {
    recipeTitle: 'My recipe',
    recipeImage: 'https://spoonacular.com/cdn/ingredients_100x100/vanilla-extract.jpg',
    readyInMinutes: 20,
    ingredients:['apple', 'orange'],
    recipe: {
      //recipe:{
      name: 'My recipe',
      steps:['1.Step one', '2.step two']
      //},
    },
  },
};

it('renders correctly', () => {
  const tree = renderer
    .create(<MealRecipe
      details={details.details}
      //recipeImage={details.recipeImage}
      //readyInMinutes={details.readyInMinutes}
      //ingredients={details.ingredients}
      //recipes={details.recipes}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
