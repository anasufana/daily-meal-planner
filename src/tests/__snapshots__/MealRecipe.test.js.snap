// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="recipe-page-container slide-in-1"
>
  <a
    className="back-link"
    href="/results"
    onClick={[Function]}
  >
    &lt; Back to results
  </a>
  <h1
    className="recipe-title-page"
  >
    My recipe
  </h1>
  <div
    className="section-ingredients"
  >
    <div
      className="recipe-img-box"
    >
      <img
        alt="test-img"
        className="image-recipe-page"
        src="https://spoonacular.com/recipeImages/Salmon-Artichoke-Quiche-516473.png"
      />
      <div
        className="clock-div-container"
      >
        <div
          className="clock-div-button"
        >
          <img
            alt="Time"
            className="img-clock"
            height="20"
            src="/img/clock.svg"
            title="Time"
            width="20"
          />
          <h3
            className="recipe-duration-page"
          >
            95
             '
          </h3>
        </div>
      </div>
    </div>
    <div
      className="ingredients-box"
    >
      <h2
        className="ingredients-title"
      >
        Ingredients:
      </h2>
      <ul
        className="ingredients-list"
      >
        <li
          className="ingredients-list-item"
        >
          granulated sugar
        </li>
        <li
          className="ingredients-list-item"
        >
          baking powder
        </li>
        <li
          className="ingredients-list-item"
        >
          vanilla extract
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div>
      <h1
        className="h1-recipe-steps"
      >
        Recipe Steps
      </h1>
      <div
        className="steps-container"
      >
        <h2
          className="step-title"
        >
          Step 
          <span
            className="step-number"
          >
            1
          </span>
          :
        </h2>
        <p
          className="step-description"
        >
          Preheat the oven to 200 degrees F.
        </p>
      </div>
      <div
        className="steps-container"
      >
        <h2
          className="step-title"
        >
          Step 
          <span
            className="step-number"
          >
            2
          </span>
          :
        </h2>
        <p
          className="step-description"
        >
          Whisk together the flour, pecans, granulated sugar, light brown sugar, baking powder, baking soda, and salt in a medium bowl.
        </p>
      </div>
    </div>
    <div>
      <h1
        className="h1-recipe-steps"
      >
        Bourbon Molasses Butter
      </h1>
      <div
        className="steps-container"
      >
        <h2
          className="step-title"
        >
          Step 
          <span
            className="step-number"
          >
            1
          </span>
          :
        </h2>
        <p
          className="step-description"
        >
          Combine the bourbon and sugar in a small saucepan and cook over high heat until reduced to 3 tablespoons, remove and let cool.
        </p>
      </div>
      <div
        className="steps-container"
      >
        <h2
          className="step-title"
        >
          Step 
          <span
            className="step-number"
          >
            2
          </span>
          :
        </h2>
        <p
          className="step-description"
        >
          Put the butter, molasses, salt and cooled bourbon mixture in a food processor and process until smooth.
        </p>
      </div>
    </div>
  </div>
</div>
`;
